File: JAVA/FatorialSimples.java
/**
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package JAVA;

import java.util.Scanner;

public class FatorialSimples {

    public static void main(String[] args) {
        Scanner sc = new Scanner( System.in );
        int N = sc.nextInt();
        sc.close();

        System.out.println( simpleFactorial(N) );
	}

    /**
     * Return the factorial number.
     */
    public static int simpleFactorial(int N) {
        int result = 1;
        N++;

        while ( N > 1 ) {
            result = result * (N-1); 
            N--;
        } 

        return result;
    }
}

File: GO/FatorialSimples.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

/**
 * Return the factorial number.
 */
func simpleFactorial(N int) int {
	var result int = 1
	N++

	for N > 1 {
		result = result * (N - 1)
		N--
	}

	return result
}
